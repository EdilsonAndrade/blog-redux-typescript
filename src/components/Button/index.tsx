import React from 'react';
import { Container } from './styles';

interface IButtonProps {
  disabled: boolean;
  onClick(): void;
}
const Button: React.FC<IButtonProps> = ({ disabled, onClick, children }) => (
  <Container aria-label="button" disabled={disabled} onClick={onClick}>
    {children}
  </Container>
);

export default Button;
