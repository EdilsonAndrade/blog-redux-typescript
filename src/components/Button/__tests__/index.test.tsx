import { render } from '@testing-library/react'
import renderer from 'react-test-renderer';
import Button from '../'

describe('[BUTTON', () => {
    const buttonAction = jest.fn();
    const button = <Button
        onClick={buttonAction}
        disabled={false}>CLICK HERE</Button>;

    it('renders button correctly', () => {
        const tree = renderer
            .create(<Button disabled={false} onClick={() => { }} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render with CLICK HERE text", () => {
        const { getByText } = render(button);

        expect(getByText("CLICK HERE")).toBeInTheDocument()
    });

    it('should execute button event click', () => {
        const { getByText } = render(button);

        const component = getByText("CLICK HERE");
        component.click()

        expect(buttonAction).toHaveBeenCalled();


    })

})

