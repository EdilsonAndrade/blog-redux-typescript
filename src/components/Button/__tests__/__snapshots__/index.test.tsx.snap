// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BUTTON renders button correctly 1`] = `
<button
  aria-label="button"
  className="sc-bdfBQB kxKqVW"
  disabled={false}
  onClick={[Function]}
/>
`;
